File: src/main/java/io/xol/chunkstories/api/gui/FocusableGuiElement.java
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package io.xol.chunkstories.api.gui;

import io.xol.chunkstories.api.input.Input;
import org.jetbrains.annotations.NotNull;

/** Elements that can be focused, either by tab-ing on them, or by clicking */
public abstract class FocusableGuiElement extends GuiElement {

	protected FocusableGuiElement(@NotNull Layer layer, int width, int height) {
		super(layer, width, height);
	}

	public boolean isFocused() {
		return this.equals(getLayer().getFocusedElement());
	}

	/** When focused an element receives input from the keyboard */
	public boolean handleInput(Input input) {
		return false;
	}
}

File: src/main/java/xyz/chunkstories/api/gui/FocusableGuiElement.kt
//
// This file is a part of the Chunk Stories API codebase
// Check out README.md for more information
// Website: http://chunkstories.xyz
//

package xyz.chunkstories.api.gui

import xyz.chunkstories.api.input.Input

/** Elements that can be focused, either by tab-ing on them, or by clicking  */
abstract class FocusableGuiElement protected constructor(layer: Layer, width: Int, height: Int) : GuiElement(layer, width, height) {

    val isFocused: Boolean
        get() = this == layer.focusedElement

    /** When focused an element receives input from the keyboard  */
    open fun handleInput(input: Input): Boolean {
        return false
    }
}
